configuration/postgres/postgres.go

```diff
@@ -0,0 +1,296 @@
+		return pingErr
+	}
+
+	// check if table exists
+	exists := false
+	err = p.client.QueryRow(ctx, QueryTableExists, p.metadata.configTable).Scan(&exists)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (p *ConfigurationStore) Get(ctx context.Context, req *configuration.GetRequest) (*configuration.GetResponse, error) {
+	query, err := buildQuery(req, p.metadata.configTable)
+	if err != nil {
+		p.logger.Error(err)
+		return nil, err
+	}
+
+	rows, err := p.client.Query(ctx, query)
+	if err != nil {
+		// If no rows exist, return an empty response, otherwise return the error.
+		if err == sql.ErrNoRows {
+			return &configuration.GetResponse{}, nil
+		}
+		return nil, err
+	}
+	response := configuration.GetResponse{}
+	for i := 0; rows.Next(); i++ {
+		var item configuration.Item
+		var key string
+		var metadata []byte
+		var v = make(map[string]string)
+
+		if err := rows.Scan(key, &item.Value, &item.Version, &metadata); err != nil {
+			return nil, err
+		}
+		if err := json.Unmarshal(metadata, &v); err != nil {
+			return nil, err
+		}
+		item.Metadata = v
+		response.Items[key] = &item
+	}
+	return &response, nil
+}
+
+func (p *ConfigurationStore) Subscribe(ctx context.Context, req *configuration.SubscribeRequest, handler configuration.UpdateHandler) (string, error) {
+	subscribeID := uuid.New().String()
+	key := "listen " + p.metadata.configTable
+	// subscribe to events raised on the configTable
+	if oldStopChan, ok := p.subscribeStopChanMap.Load(key); ok {
+		close(oldStopChan.(chan struct{}))
+	}
+	stop := make(chan struct{})
+	p.subscribeStopChanMap.Store(subscribeID, stop)
+	go p.doSubscribe(ctx, req, handler, key, subscribeID, stop)
+	return subscribeID, nil
+}
+
+func (p *ConfigurationStore) Unsubscribe(ctx context.Context, req *configuration.UnsubscribeRequest) error {
+	if oldStopChan, ok := p.subscribeStopChanMap.Load(req.ID); ok {
+		p.subscribeStopChanMap.Delete(req.ID)
+		close(oldStopChan.(chan struct{}))
+	}
+	return nil
+}
+
+func (p *ConfigurationStore) doSubscribe(ctx context.Context, req *configuration.SubscribeRequest, handler configuration.UpdateHandler, channel string, id string, stop chan struct{}) {
+	conn, err := p.client.Acquire(ctx)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Error acquiring connection:", err)
+	}
+	defer conn.Release()
+
+	_, err = conn.Exec(context.Background(), channel)
+	if err != nil {
+		p.logger.Errorf("Error listening to channel:", err)
+		return
+	}
+
+	for {
+		notification, err := conn.Conn().WaitForNotification(ctx)
+		if err != nil {
+			p.logger.Errorf("Error waiting for notification:", err)
+			return
+		}
+		p.handleSubscribedChange(ctx, handler, notification, id)
+	}
+}
+
+func (p *ConfigurationStore) handleSubscribedChange(ctx context.Context, handler configuration.UpdateHandler, msg *pgconn.Notification, id string) {
+	defer func() {
+		if err := recover(); err != nil {
+			p.logger.Errorf("panic in handleSubscribedChange(）method and recovered: %s", err)
```

Comment:
Note there's a non-ASCII character here that VS Code is highlighting.

Reply:
Fixed